core/src/test/scala/doric/sem/CollectOpsSpec.scala

```diff
@@ -0,0 +1,95 @@
+package doric
+package sem
+
+import scala.reflect.ClassTag
+
+import doric.types.{CustomSparkType, SparkType}
+import doric.Equalities._
+import org.scalactic.Equality
+import org.scalatest.Assertion
+
+import org.apache.spark.sql.Encoder
+
+case class User(name: String, surname: String)
+
+class CollectOpsSpec extends DoricTestElements {
+
+  import spark.implicits._
+
+  def test[T: SparkType: Encoder](
+      l: List[T]
+  )(implicit eq: Equality[List[T]]): Assertion =
+    l.toDF("nums")
+      .collectCols(col[T](c"nums")) should ===(l)
+
+  describe("collectCols") {
+    it("allows to collect simple columns") {
+      val maybeInts = List(Option(1), None)
+      test(maybeInts)
+
+      val nullAndStrings: List[String] = List("1", null)
+      test(nullAndStrings)
+
+      val maybeStrings: List[Option[String]] = List(Some("1"), None)
+      test(maybeStrings)
+
+      val listOfStrings: List[Array[String]] =
+        List(Array("a", "b"), Array.empty[String])
+      test(listOfStrings)
+    }
+
+    it("allows to collect maps") {
+      val x = List(Map("hola" -> "adios"))
+      test(x)
+
+      val y = List(Map("hola" -> Map("jander" -> 1)))
+      test(y)
+    }
+
+    it("allows to collect custom elements") {
+      implicit def listSparkType[T: SparkType: ClassTag](implicit
+          arr: SparkType[Array[T]]
+      ): CustomSparkType[List[T]] =
+        SparkType[Array[T]].customType[List[T]](_.toArray, _.toList)
+
+      spark
+        .range(1)
+        .select(List(1, 2, 3).lit.as(c"value"))
+        .collectCols(col[List[Int]](c"value")) shouldBe List(List(1, 2, 3))
+
+      implicit val userst: CustomSparkType[User] =
+        SparkType[String].customType[User](
+          x => s"${x.name}#${x.surname}",
+          x => {
+            val name :: surname :: Nil = x.split("#").toList
+            User(name, surname)
+          }
+        )
+
+      val user = User("name", "surname")
+      spark
+        .range(1)
+        .select(user.lit.as(c"value"))
+        .collectCols(col[User](c"value")) shouldBe List(user)
+
+      spark
+        .range(1)
+        .select(List(user).lit.as(c"value"))
+        .collectCols(col[List[User]](c"value")) shouldBe List(List(user))
+
+      spark
```

Comment:
shouldn't them be multiple tests?

Reply:
si not a test for user, is a test for custom types in general, but we can split them yes